src/App.tsx

```diff
@@ -15,48 +16,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Name = 'name',
+  Length = 'length',
+  Default = '',
+}
+
+function sortGoods(goods: string[], sortType: SortType, isReversed: boolean) {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        case SortType.Name:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState<SortType>(SortType.Default);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = sortGoods(goodsFromServer, sortType, isReversed);
+  const reset = () => {
```

Comment:
Could you please separate all hooks, vars definitions, and functions with empty lines, it will improve code readability.

Reply:
➕